lib/stackify-api-ruby.rb

```diff
@@ -103,25 +103,27 @@ def internal_log level, msg
       Stackify.logger.send(level.downcase.to_sym, Stackify::INTERNAL_LOG_PREFIX){ msg }
     end
 
-    def run async = true
-      if Stackify.is_valid?
-        at_exit { make_remained_job }
-         t1 = Thread.new { Stackify.authorize }
-        case Stackify.configuration.mode
-        when MODES[:both]
-          t2 = start_logging
-          t3 = start_metrics
-        when MODES[:logging]
-          t2 = start_logging
-        when MODES[:metrics]
-          t3 = start_metrics
-        end
-        unless async
+    def run
+      Stackify::Utils.is_api_enabled
+      if Stackify.configuration.api_enabled
+        if Stackify.is_valid?
+          at_exit { make_remained_job }
+           t1 = Thread.new { Stackify.authorize }
+          case Stackify.configuration.mode
+          when MODES[:both]
+            t2 = start_logging
+            t3 = start_metrics
+          when MODES[:logging]
+            t2 = start_logging
+          when MODES[:metrics]
+            t3 = start_metrics
+          end
+
           t1.join
-          t2.join if t2
+          t3.join if t3
```

Comment:
why is t2 not joined ?

Reply:
I remember that we have the authorization error occured in Passenger so part of the fix is not to join the t2. I already tried it without running a Passenger and still the logging of stackify api works fine.